RacingMod/Chat/RacingCommands.cs

```diff
@@ -0,0 +1,201 @@
+            }
+        }
+
+        public void Unload()
+        {
+            MyAPIGateway.Utilities.MessageEntered -= MessageEntered;
+            MyAPIGateway.Utilities.MessageRecieved -= ReceiveCommand;
+        }
+
+        // Called on server
+        private void ReceiveCommandPacket(byte[] data)
+        {
+            try
+            {
+                CommandInfo cmd = MyAPIGateway.Utilities.SerializeFromBinary<CommandInfo>(data);
+                ReceiveCommand(cmd.id, cmd.command);
+            }
+            catch (Exception e)
+            {
+                RacingTools.ShowError(e, GetType());
+            }
+        }
+
+        // Called on server
+        private void ReceiveCommand(ulong id, string command)
+        {
+            IMyPlayer p = RacingTools.GetPlayer(id);
+            if (p != null)
+                ProcessCommand(p, command);
+        }
+
+        // Called on clients
+        private void MessageEntered(string messageText, ref bool sendToOthers)
+        {
+            IMyPlayer p = MyAPIGateway.Session.Player;
+            if (p == null)
+                return;
+            
+            if (IsRaceCommand(messageText))
+            {
+                sendToOthers = false;
+
+                if (RacingConstants.IsServer)
+                    ProcessCommand(p, messageText);
+                else
+                    RacingSession.Instance.Net.SendToServer(RacingConstants.packetCmd, new CommandInfo(messageText, p.SteamUserId));
+            }
+        }
+
+        private bool IsRaceCommand(string command)
+        {
+            return command.StartsWith("/rcd", StringComparison.OrdinalIgnoreCase) || command.StartsWith(prefix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        private void ProcessCommand(IMyPlayer p, string command)
+        {
+            if (!IsRaceCommand(command))
+                return;
+
+            bool admin = RacingTools.IsPlayerAdmin(p);
+            string[] cmd = command.ToLowerInvariant().Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+            ChatCommand cmdOb;
+            if (cmd.Length <= 1 || !commands.TryGetValue(cmd[1], out cmdOb))
+                ShowChatHelp(p, admin);
+            else
+                cmdOb.Run(p, admin, cmd, race);
+        }
+
+        private void BuildHelpText()
+        {
+            // Admin
+            StringBuilder admin = new StringBuilder("Admin Commands:");
+            HashSet<string> printed = new HashSet<string>();
+            foreach (ChatCommand cmd in commands.Values)
+            {
+                AdminCommand acmd = cmd as AdminCommand;
+                if (acmd != null && printed.Add(cmd.Id))
+                {
+                    admin.AppendLine();
+                    AppendCommand(admin, acmd.Id, acmd.Usage);
+                }
+            }
+
+            string adminHelp = admin.ToString();
+
```

Comment:
Readability: Long function, split into two along the loops. Like BuildAdminHelpText() and BuildGeneralHelpText() or something similar.